include/mqtt/endpoint.hpp

```diff
@@ -8278,1217 +8325,3808 @@ class endpoint : public std::enable_shared_from_this<endpoint<Socket, Mutex, Loc
+                    );
+                    process_property_id(
+                        std::move(func),
+                        std::move(buf),
+                        rest,
+                        std::move(props),
+                        std::move(handler),
+                        std::move(self)
+                    );
+                },
+                std::move(self)
+            );
+        } break;
+        case v5::property::id::subscription_identifier_available: {
+            std::size_t const len = 1;
+            if (property_length_rest < len) {
+                call_message_size_error_handlers(func);
+                return;
+            }
+            process_nbytes(
+                std::move(func),
+                std::move(buf),
+                len,
+                [
+                    this,
+                    props = std::move(props),
+                    handler = std::move(handler),
+                    rest = property_length_rest - len
+                ]
+                (buffer body, buffer buf, async_handler_t func, this_type_sp self) mutable {
+                    props.emplace_back(
+                        v5::property::subscription_identifier_available(body.begin(), body.end())
+                    );
+                    process_property_id(
+                        std::move(func),
+                        std::move(buf),
+                        rest,
+                        std::move(props),
+                        std::move(handler),
+                        std::move(self)
+                    );
+                },
+                std::move(self)
+            );
+        } break;
+        case v5::property::id::shared_subscription_available: {
+            std::size_t const len = 1;
+            if (property_length_rest < len) {
+                call_message_size_error_handlers(func);
+                return;
+            }
+            process_nbytes(
+                std::move(func),
+                std::move(buf),
+                len,
+                [
+                    this,
+                    props = std::move(props),
+                    handler = std::move(handler),
+                    rest = property_length_rest - len
+                ]
+                (buffer body, buffer buf, async_handler_t func, this_type_sp self) mutable {
+                    props.emplace_back(
+                        v5::property::shared_subscription_available(body.begin(), body.end())
+                    );
+                    process_property_id(
+                        std::move(func),
+                        std::move(buf),
+                        rest,
+                        std::move(props),
+                        std::move(handler),
+                        std::move(self)
+                    );
+                },
+                std::move(self)
+            );
+        } break;
+        }
+    }
+
+    // process common
+
+    templat
```

Comment:
How practical would it be to make all of these process_*** functions into free functions, that accessed the member variables of the endpoint object via the passed in std::shared_ptr<endpoint<>> parameter?

Reply:
I think that the code reference is something wrong. Could you update it?

Reply:
The code reference is wrong, but I was talking about `process_connect`, and `process_pingresp` and similar.

Reply:
Some of proces_* functions access `buf_` member variable to avoid allocation. It is a private member of endpoint.
For example https://github.com/redboltz/mqtt_cpp/blob/impl_320_2/include/mqtt/endpoint.hpp#L8487-L8505

I think that they should be member function and not practical to become free function. I think that frined function with endpoint parameter is worse than member function.

Reply:
Ok. I understand.